ui/admin/app/controllers/scopes/scope/workers/worker/tags.js

```diff
@@ -4,10 +4,75 @@
  */
 
 import Controller, { inject as controller } from '@ember/controller';
-import { TAG_TYPE_CONFIG } from 'api/models/worker';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { loading } from 'ember-loading';
+import { notifySuccess, notifyError } from 'core/decorators/notify';
+import { TAG_TYPE_CONFIG, TAG_TYPE_API } from 'api/models/worker';
+import { TrackedObject } from 'tracked-built-ins';
 
 export default class ScopesScopeWorkersWorkerTagsController extends Controller {
   @controller('scopes/scope/workers/index') workers;
 
+  // =services
+  @service intl;
+
   tagTypeConfig = TAG_TYPE_CONFIG;
+  tagTypeApi = TAG_TYPE_API;
+
+  modals = new TrackedObject({
+    remove: false,
+    edit: false,
+  });
+  @tracked removalConfirmation = null;
+  @tracked apiTagToRemove = null;
+
+  /**
+   * Build the display name for a worker tag
+   * @returns {string}
+   */
+  get tagDisplayName() {
+    return `"${this.apiTagToRemove?.key} = ${this.apiTagToRemove?.value}"`;
+  }
+
+  /**
+   * Checks if user entered REMOVE to enable the remove button.
+   * @returns {boolean}
+   */
+  get isRemovalConfirmed() {
+    return (
+      this.removalConfirmation === this.intl.t('actions.remove').toUpperCase()
+    );
+  }
+
+  /**
+   * Removes selected tag from the worker.
+   */
+  @action
+  @loading
+  @notifyError(({ message }) => message, { catch: true })
+  @notifySuccess('resources.worker.tags.remove.success')
+  async removeApiTag() {
+    await this.model.removeApiTags({
+      [this.apiTagToRemove.key]: [this.apiTagToRemove.value],
+    });
+    this.modals.remove = false;
+    this.apiTagToRemove = null;
+    this.removalConfirmation = null;
+  }
+
+  /**
+   * Opens the remove tag modal.
+   * @param {string} action
+   * @param {object} apiTag
+   */
+  @action
+  toggleModal(action, apiTag = null) {
+    if (action === 'remove') {
+      this.modals.remove = !this.modals.remove;
+      this.apiTagToRemove = apiTag;
+      this.removalConfirmation = null;
```

Comment:
do we need to set `removalConfirmation`  to null in this action?  we set this to `null` on page load and we only reassign on input change, so I wonder if this is needed here

Reply:
I added it here because the user can open the remove modal, type something in then cancel or close the modal. Now if they open the modal again the same input will still be there.